tls/extensions/s2n_client_server_name.c

```diff
@@ -44,3 +44,45 @@ int s2n_extensions_client_server_name_send(struct s2n_connection *conn, struct s
 
     return 0;
 }
+
+int s2n_parse_client_hello_server_name(struct s2n_connection *conn, struct s2n_stuffer *extension)
+{
+    if (conn->server_name[0]) {
+        /* already parsed server name extension, exit early */
+        return 0;
+    }
+
+    uint16_t size_of_all;
+    uint8_t server_name_type;
+    uint16_t server_name_len;
+    uint8_t *server_name;
+
+    GUARD(s2n_stuffer_read_uint16(extension, &size_of_all));
+    if (size_of_all > s2n_stuffer_data_available(extension) || size_of_all < 3) {
```

Comment:
same as above: Define a constant value for `3` here.

Reply:
This is just moving the logic over. from one file to another. removal of magic numbers will come at a later point in optimizing